### PR TITLE
Add requirements-dev.txt

### PR DESCRIPTION
**Pull request by bopardikarsoham**
I have added a `requirements-dev.txt` file as per Issue https://github.com/Qiskit/qiskit-tutorials/issues/1317. This contains all the packages used in the `qiskit-tutorials` repo. I also double-checked all of the packages required for this repository.

### CODE REVIEW

**Comment by HuangJunye** [comment]
Thanks! Can you try to do a local build with sphinx with a fresh environment to check that this file contains all and only necessary packages?

**Comment by bopardikarsoham** [comment]
Sure, I'll test it out locally

**Comment by bopardikarsoham** [comment]
@HuangJunye I have tested it locally by creating a new virtual environment in my PC and installing all the packages from scratch in the requirements-dev.txt file through the pip install -r requirements-dev.txt. The packages were successfully installed and everything seems to be working including sphinx thumbnails.

**Comment by HuangJunye** [comment]
@bopardikarsoham That's awesome! I am having some troubles do a local build on my machine. https://github.com/Qiskit/qiskit-tutorials/issues/1324 

I think a lot of these requirements are already in qiskit and qiskit[visualization]. Can you try this simplified version?

```
qiskit
jupyter
sphinx
nbsphinx
sphinx_rtd_theme
networkx
scikit-learn
matplotlib
qiskit[visualization]
cvxpy
pyscf
graphviz
```
I had a look at the CI and I think these should be sufficient based on https://github.com/Qiskit/qiskit-tutorials/blob/c98b29af9ca4350a940f5e48b911ffb4f1be5557/azure-pipelines.yml#L39-L43 The only missing requirement would be pandoc but it's not pip installable because it's not a python package.

**Comment by bopardikarsoham** [comment]
Yes @HuangJunye , I believe this list is sufficient and capable of installing all of the packages required for this repo to function. I figured that out too in the pipelines file.

**Comment by bopardikarsoham** [comment]
> @bopardikarsoham That's awesome! I am having some troubles making a local build on my machine. #1324
> 
> I think a lot of these requirements are already in qiskit and qiskit[visualization]. Can you try this simplified version?
> 
> ```
> qiskit
> jupyter
> sphinx
> nbsphinx
> sphinx_rtd_theme
> networkx
> scikit-learn
> matplotlib
> qiskit[visualization]
> cvxpy
> pyscf
> graphviz
> ```
> 
> I had a look at the CI and I think these should be sufficient based on
> 
> https://github.com/Qiskit/qiskit-tutorials/blob/c98b29af9ca4350a940f5e48b911ffb4f1be5557/azure-pipelines.yml#L39-L43
> 
> The only missing requirement would be pandoc but it's not pip installable because it's not a python package.

Replacing `graphviz` with `pydot` will fix everything

**Comment by mtreinish** [comment]
graphviz isn't a python dependency and can not be listed in a requirements list as pip will not be able to install it for you. It is something you have to install separately through a different mechanism (see: https://graphviz.org/download/ )

**Review by HuangJunye — CHANGES_REQUESTED** [review]
I made a suggestion to change. I tested on macOS and it's working together with conda install pandoc and graphviz.

File: requirements-dev.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+wheel
+pylatexenc>=1.4
+ddt>=1.2.0,!=1.4.0,!=1.4.3
+reno>=3.4.0
+pygments>=2.4
+scikit-learn>=0.20.0
+scikit-quant<=0.7;platform_system != 'Windows'
+jax;platform_system != 'Windows'
+jaxlib;platform_system != 'Windows'
+docplex
```
[the remark this answers is not in the document]
Comment (bopardikarsoham) [reply requirements-dev.txt]: Yes I have updated the PR accordingly!

**Review by HuangJunye — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**